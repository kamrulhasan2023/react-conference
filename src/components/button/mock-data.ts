import {
  IFillButton,
  IFillRedirectButton,
  IOutlineButton,
  IOutlineRedirectButton,
} from './interface';

export const buttonProps: IFillButton = {
  label: 'Button',
  colorScheme: 'bg-primary-text-white',
  hoverColorScheme: 'bg-success-text-white',
  size: 'lg',
  clickHandler: () => {},
  variant: 'fill',
};

export const redirectButtonProps: IFillRedirectButton = {
  href: 'https://www.google.com/',
  isOpenNewTab: true,
  label: 'Redirect Button',
  colorScheme: 'bg-primary-text-white',
  hoverColorScheme: 'bg-success-text-white',
  size: 'lg',
  variant: 'fill',
};

export const outlineButtonProps: IOutlineButton = {
  label: 'Outline Button',
  colorScheme: 'bg-transparent-text-primary',
  hoverColorScheme: 'bg-transparent-text-success',
  size: 'lg',
  clickHandler: () => {},
  variant: 'outline',
};

export const redirectOutlineButtonProps: IOutlineRedirectButton = {
  href: 'https://www.google.com/',
  isOpenNewTab: true,
  label: 'Outline Redirect Button',
  colorScheme: 'bg-transparent-text-primary',
  hoverColorScheme: 'bg-transparent-text-success',
  size: 'lg',
  variant: 'outline',
};
