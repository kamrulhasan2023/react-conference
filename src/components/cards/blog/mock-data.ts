import { IBlogCard } from './interface';

export const blogCardProps: IBlogCard = {
  href: 'https://www.lemonhive.com/',
  image: {
    src: 'https://res.cloudinary.com/dktw59lc2/image/upload/v1669267967/BrightonSEO/Carousel/image-2_yph8q6.jpg',
    alt: 'dummy image 2',
    lqip: 'https://res.cloudinary.com/dktw59lc2/image/upload/v1669267967/BrightonSEO/Carousel/image-2_yph8q6.jpg',
  },
  title: 'Headless Website Development',
  description:
    'Headless solutions built with Sanity.io, Strapi, Prismic, Contentful and others with front-end frameworks such as Next.js, Gatsby, & Nuxt.',
  tags: [
    { fontColor: { hex: 'black' }, bgColor: { hex: 'red' }, title: 'Reactjs' },
    { fontColor: { hex: 'black' }, bgColor: { hex: 'red' }, title: 'Nextjs' },
  ],
  isFeatured: false,
};
