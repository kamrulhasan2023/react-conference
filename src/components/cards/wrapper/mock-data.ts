import { ICardWrapper } from './interface';

export const cardWrapperProps: ICardWrapper = {
  cardBorderRadius: 'radius-none',
  cardShadow: 'shadow-one',
};
