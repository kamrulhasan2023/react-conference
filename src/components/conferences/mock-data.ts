import {IConferences} from "./interface";
export const conferencesProps:IConferences = {
    title : "Conferences",
    conferences: [
        {   
            id:1,
            title: "Freezing Edge 2023",
            description:"The edge isn't bleeding, it's freezing!",
            startDate: "02 September, 2023",
        },
        {
            id:2,
            title: "Freezing Edge 2023",
            description:"The edge isn't bleeding, it's freezing!",
            startDate: "02 September, 2023",
        },
        {
            id:3,
            title: "Freezing Edge 2023",
            description:"The edge isn't bleeding, it's freezing!",
          startDate: "02 September, 2023",
        },
        {
            id:4,
            title: "Freezing Edge 2023",
            description:"The edge isn't bleeding, it's freezing!",
            startDate: "02 September, 2023",
           
        },
        {
            id:5,
            title: "Freezing Edge 2023",
            description:"The edge isn't bleeding, it's freezing!",
            startDate: "02 September, 2023",
        }
    ]

}