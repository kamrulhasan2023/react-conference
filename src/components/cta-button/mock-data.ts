import { ICtaButton } from './interface';

export const ctaButtonProps: ICtaButton = {
  label: 'Cta button',
  href: '/',
  colorScheme: 'primary',
  isOpenNewTab: false,
};
