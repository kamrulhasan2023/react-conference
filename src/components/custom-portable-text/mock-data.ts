import { ISanityCustomPortableText } from 'src/backend/queries/portable-text/interface';
import { ICustomPortableText } from './interface';

export const richTextContent: ISanityCustomPortableText = [
  {
    _key: '88cb73b1f4cf',
    _type: 'block',
    children: [
      {
        _key: 'b07265ab9f080',
        _type: 'span',
        marks: ['textColorOne'],
        text: '#1 - Lorem Ipsum is simply',
      },
    ],
    markDefs: [],
    style: 'headingOne',
  },
  {
    _key: '521b4d80be61',
    _type: 'block',
    children: [
      {
        _key: '112402beef9e0',
        _type: 'span',
        marks: ['d6eda9d200d9'],
        text: 'Sanity',
      },
      { _key: '19ad07fbf130', _type: 'span', marks: [], text: ' ' },
      {
        _key: '9dde6d5f4dae',
        _type: 'span',
        marks: ['728a960cf447'],
        text: 'offers',
      },
      { _key: '88955872dc8e', _type: 'span', marks: [], text: ' ' },
      {
        _key: 'faac8da27e67',
        _type: 'span',
        marks: [],
        text: 'a range of digital solutions for online businesses, ',
      },
      {
        _key: '1b7e6ac8ee6b',
        _type: 'span',
        marks: [],
        text: 'from e-commerce storefronts to marketing sites and innovative mobile apps with fast structured content. Sanity’s goal is to help businesses improve the quality of their digital products and services and reduce their Time to Market.',
      },
    ],
    markDefs: [
      {
        _key: 'd6eda9d200d9',
        _type: 'link',
        href: 'https://www.sanity.com',
        linkType: 'externalLink',
      },
      {
        _key: '728a960cf447',
        _type: 'link',
        internalLink: { slug: { _type: 'slug', current: '/' } },
        linkType: 'internalLink',
      },
    ],
    style: 'normal',
  },
  {
    _key: 'ea58243269f4',
    _type: 'block',
    children: [
      {
        _key: '7b48c62108af0',
        _type: 'span',
        marks: ['highlighColorOne'],
        text: 'Tools and solutions introduced by Sanity',
      },
      {
        _key: '582b37ee33b9',
        _type: 'span',
        marks: [],
        text: ' create a unified and developer-friendly content management environment, based on modern frameworks and flawless API integration. Solutions like headless commerce and headless CMS offer supreme customizability, scalability, and flexibility, and Sanity makes it simple and stress-free to integrate and deploy them for online businesses.',
      },
    ],
    markDefs: [],
    style: 'normal',
  },
  {
    _key: '1e8a7958e08b',
    _type: 'block',
    children: [
      {
        _key: '891dafc015800',
        _type: 'span',
        marks: [],
        text: 'The platform offers convenient document querying through GraphQL and GROQ, as well as simple data management with structured content, with all the complicated work done for you over a cloud-hosted service - and on top of all, there’s a massive community of developers and creators from all over the world sharing their advice, tools, libraries, and more!',
      },
    ],
    markDefs: [],
    style: 'normal',
  },
  {
    _key: 'b6b1427a71b9',
    _type: 'block',
    children: [{ _key: '79b1538b7440', _type: 'span', marks: [], text: '' }],
    markDefs: [],
    style: 'normal',
  },
  {
    _key: '9c78e641cbfa',
    _type: 'customImageSchema',
    altDescription: 'Image',
    imageFile: {
      asset: {
        metadata: {
          dimensions: { aspectRatio: 1.7875, height: 640, width: 1144 },
          lqip: 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAALCAYAAAB/Ca1DAAAACXBIWXMAAAsTAAALEwEAmpwYAAABHElEQVQokY2SSY6EQAwE+f9/uCNewQkuSAjETrNDuRWecYvhMDRSquyqdHrD67pO5nmW8zzFOfcIeMdxKLD51nWVJEkkDEPx9n3/PH4jtiyLtG0rwzAIsc45je/7XrIsE+9Kvtrmm03Qtm0qRGBZljJN06cY46ggDmQDIxjHUVsB2AABzrquhVHhL8uisGpVEOf1eikgVVUleZ5rNbRSFIXe8U4y7uGBvu+laRp9ozAVpCouyE42IxHMPe3hW4U2Q/xhGNSH96dlqgR3m2QEIk77BFKxJdx+x2R875vNXoEIM7xvues6SdP0WfAOghkLVdl/iB9Fkfi+//MfXjd8bcHa+C8BH/w4jiUIAvEYsm2TE9iCmB3kJ0FOm+8bo7BaYW2eHzAAAAAASUVORK5CYII=',
        },
        url: 'https://cdn.sanity.io/images/9rg5iv1m/development/b6439bd820dcc42c4bf24461042ac2905f757ee3-1144x640.png',
      },
      crop: null,
      hotspot: null,
    },
    markDefs: null,
  },
  {
    _key: '984f842a122e',
    _type: 'block',
    children: [
      {
        _key: '712e3ef40b9e0',
        _type: 'span',
        marks: ['textColorTwo'],
        text: 'Lorem Ipsum is simply dummy',
      },
    ],
    markDefs: [],
    style: 'headingTwo',
  },
  {
    _key: '94c98802b7d5',
    _type: 'block',
    children: [
      {
        _key: 'af722fac9a4f',
        _type: 'span',
        marks: ['textColorTwo'],
        text: 'Title 2',
      },
    ],
    markDefs: [],
    style: 'headingTwo',
  },
  {
    _key: '0fc9921c004f',
    _type: 'block',
    children: [
      {
        _key: '8eefa2c99dde0',
        _type: 'span',
        marks: [],
        text: 'Lorem ipsum dolor sit amet consectetur. Laoreet euismod dignissim et pellentesque sed egestas nec. Viverra urna pretium tortor vel blandit mattis. Et quis diam mi augue arcu enim. Magna placerat potenti consectetur eget eu nibh. Nunc semper gravida orci nulla tellus volutpat. Orci felis arcu faucibus at. Amet accumsan vitae in malesuada vestibulum. Ligula risus velit risus dolor fermentum nunc quisque quis.',
      },
    ],
    markDefs: [],
    style: 'normal',
  },
  {
    _key: '6861203dfa82',
    _type: 'block',
    children: [
      {
        _key: 'a8abb452f3d90',
        _type: 'span',
        marks: [],
        text: 'Eu sagittis tristique pellentesque sed. Donec massa curabitur ut egestas tristique ac elementum. Nphasellus turpis id lacus ornare dui purus turpis sit. Ligula ac cras aliquam velit. Fermentum est volutpat enim odio leo massa. Eget ullamcorper nunc arcu magnis vitae nunc vitae pharetra.',
      },
    ],
    markDefs: [],
    style: 'normal',
  },
  {
    _key: '3f72ecb8ea21',
    _type: 'block',
    children: [{ _key: 'cd7a9bcc7fb5', _type: 'span', marks: [], text: '' }],
    markDefs: [],
    style: 'normal',
  },
  {
    _key: 'f0154f9aea46',
    _type: 'block',
    children: [
      {
        _key: 'ae5d012b0ade0',
        _type: 'span',
        marks: ['textColorOne'],
        text: '#2 - Lorem Ipsum is simply dummy',
      },
    ],
    markDefs: [],
    style: 'headingOne',
  },
  {
    _key: '0fde70cc9b89',
    _type: 'blockquote',
    content: [
      {
        _key: '746e53098e03',
        _type: 'block',
        children: [
          {
            _key: 'c7331cd56b050',
            _type: 'span',
            marks: ['em'],
            text: 'Lorem ipsum dolor sit amet consectetur. Laoreet euismod digniss et pellentesque sed egestas nec. Viverra urna pretium tortor vel blandit mattis. Et quis diam mi augue arcu enim. Magna placeraconsectetur eget eu nibh. Nunc semper gravida orci nulla tellus volutpat. Orci felis arcu faucibus at.',
          },
        ],
        markDefs: [],
        style: 'normal',
      },
    ],
    editorTitle: 'Anonymous',
    markDefs: null,
  },
  {
    _key: 'e7700421184c',
    _type: 'block',
    children: [
      {
        _key: 'c93289e815020',
        _type: 'span',
        marks: [],
        text: 'Lorem ipsum dolor sit amet consectetur. Laoreet euismod dignissim et pellentesque sed egestas nec. Viverra urna pretium tortor vel blandit mattis. Et quis diam mi augue arcu enim. Magna placerat potenti consectetur eget eu nibh. Nunc semper gravida orci nulla tellus volutpat. Orci felis arcu faucibus at. Amet accumsan vitae in malesuada vestibulum. Ligula risus velit risus dolor fermentum nunc quisque quis.',
      },
    ],
    markDefs: [],
    style: 'normal',
  },
  {
    _key: 'a09da5cde0ed',
    _type: 'block',
    children: [
      {
        _key: '605680390ec50',
        _type: 'span',
        marks: [],
        text: 'Eu sagittis tristique pellentesque sed. Donec massa curabitur ut egestas tristique ac elementum. Nphasellus turpis id lacus ornare dui purus turpis sit. Ligula ac cras aliquam velit. Fermentum est volutpat enim odio leo massa. Eget ullamcorper nunc arcu magnis vitae nunc vitae pharetra.',
      },
    ],
    markDefs: [],
    style: 'normal',
  },
  {
    _key: 'bff38b041985',
    _type: 'block',
    children: [
      {
        _key: '8b9c0213b3f90',
        _type: 'span',
        marks: ['textColorThree'],
        text: 'Lorem Ipsum is simply dummy',
      },
    ],
    markDefs: [],
    style: 'headingThree',
  },
  {
    _key: '015f68db76bf',
    _type: 'block',
    children: [
      {
        _key: '37de860847560',
        _type: 'span',
        marks: [],
        text: 'Sanity Connect is a powerful Shopify app that allows e-commerce site owners to create beautiful and fast-performing storefronts, designed and engineered by Sanity.io - the creators of a flexible, unified content platform built to create custom-made digital solutions based on React, Gatsby, Next, Vue, and more.',
      },
    ],
    markDefs: [],
    style: 'normal',
  },
  {
    _key: '9af88f2d81ca',
    _type: 'block',
    children: [
      {
        _key: '7b9df7c4d6ef0',
        _type: 'span',
        marks: [],
        text: 'To attract new customers and retain those who have already made purchases, online companies need to rely on exceptional content. With the Sanity Connect app, you can install, manage, and edit your page’s content in real-time, using a simple and user-friendly UI that holds a range of powerful features. Sanity is an official partner of Shopify, and the two platforms cooperate to create the best storefront solutions.',
      },
    ],
    markDefs: [],
    style: 'normal',
  },
  {
    _key: 'ab66fc43fc01',
    _type: 'block',
    children: [{ _key: '9e08862a2ac3', _type: 'span', marks: [], text: '' }],
    markDefs: [],
    style: 'normal',
  },
  {
    _key: 'a38f6b962a43',
    _type: 'customImageSchema',
    altDescription: 'Image 2',
    imageFile: {
      asset: {
        metadata: {
          dimensions: { aspectRatio: 1.6875, height: 640, width: 1080 },
          lqip: 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAMCAYAAABiDJ37AAAACXBIWXMAAAsTAAALEwEAmpwYAAACL0lEQVQokW2T3YtTSRBH7/8v+L745Iuwiw+ruMgOLIIiiii4iijDxJnJTDJJTPp+9HdXd450J+qK+3Dognuo29SvuhNJWGOw1pCSsN/v/5eYElrrRvCeItUtv3idiGCMRRuHD4mQMvE/JMnkUkjOohdXmNkp4ewTaX5OmqaDJwXJhVL2dDkXjIsM2jPowGAq8TvapdZU1Bb38inmz3vY+3fRJw8YZp9Q/ZbNOKKMI0qmyylhdor+eom6ukbVc7VGfdmilGKqt4iRvLnBnzzA3LmN/u0Wm9/vMH/1mM9nz3h1/i9vVlcob+lK8PirS8yHd+j3bzEf32E+f0Rfn2LWM1y/QIIh9zvC6+e4v+5jH/3B+p+HXLz9m9npCU/PXvJ8OWPjNF1JiVjl1YK4WiJfVohaI+MG0Ruy6ykSyCEQtxvC9SVxMcetF2i1ZBwXrMYNazvhJdFJFpxzeOeRJJQ63EamlEpNspDk4FXaCNr3cmRP+ZZyFY3zaBuwIeFixsfczkpNvSZY18bYo+cP3gHBR2lpt5Rjkib12h8JPzHZeFwhYbQeNRr6oWfotwzDQD+55mkX24+7usyTDewmx3ayR37Ug/GtmYuRxThwsZ6zvXzNdP6McfmBXa+aP9rQ1qsTyRhfd8/RG9fOb3VF+yoKJgbO+y2nN2fs5i8IF0+wN+8ZR8VgXeshOdPVoaZcZ1VfSSKIEORYp7rU0l6KlIyOntFNBKfIbof4+lLCwTuG9BW4RZUQzuUIkAAAAABJRU5ErkJggg==',
        },
        url: 'https://cdn.sanity.io/images/9rg5iv1m/development/21adc771531fee87fddd5ba03d99069478539c8e-1080x640.png',
      },
      hotspot: null,
      crop: null,
    },
    markDefs: null,
  },
  {
    _key: '9b42844b72c7',
    _type: 'block',
    children: [
      {
        _key: 'a451c45b18ad0',
        _type: 'span',
        marks: ['textColorOne'],
        text: '#3 - Lorem Ipsum',
      },
    ],
    markDefs: [],
    style: 'headingOne',
  },
  {
    _key: '1078fc1db900',
    _type: 'block',
    children: [
      {
        _key: 'a02c9f46de670',
        _type: 'span',
        marks: [],
        text: 'Lorem ipsum dolor sit amet consectetur. Laoreet euismod dignissim et pellentesque sed egestas nec. Viverra urna pretium tortor vel blandit mattis. Et quis diam mi augue arcu enim. Magna placerat potenti consectetur eget eu nibh. Nunc semper gravida orci nulla tellus volutpat. Orci felis arcu faucibus at. Amet accumsan vitae in malesuada vestibulum. Ligula risus velit risus dolor fermentum nunc quisque quis.',
      },
    ],
    markDefs: [],
    style: 'normal',
  },
  {
    _key: '89e1281f8d43',
    _type: 'block',
    children: [
      {
        _key: '67612d111c890',
        _type: 'span',
        marks: [],
        text: 'Eu sagittis tristique pellentesque sed. Donec massa curabitur ut egestas tristique ac elementum. Nphasellus turpis id lacus ornare dui purus turpis sit. Ligula ac cras aliquam velit. Fermentum est volutpat enim odio leo massa. Eget ullamcorper nunc arcu magnis vitae nunc vitae pharetra.',
      },
    ],
    markDefs: [],
    style: 'normal',
  },
  {
    _key: '514d41bda93d',
    _type: 'block',
    children: [{ _key: '57034390f577', _type: 'span', marks: [], text: '' }],
    markDefs: [],
    style: 'normal',
  },
  {
    _key: 'b6d752a21f81',
    _type: 'block',
    children: [
      {
        _key: 'a2b3960ded69',
        _type: 'span',
        marks: ['textColorOne'],
        text: '#4 - Reel Video',
      },
    ],
    markDefs: [],
    style: 'headingOne',
  },
  {
    _key: '0121abc30603',
    _type: 'customVideoSchema',
    altDescription: 'Tiinloof',
    imageFile: {
      asset: {
        metadata: {
          dimensions: {
            aspectRatio: 1.5711947626841243,
            height: 1222,
            width: 1920,
          },
          lqip: 'data:image/jpeg;base64,/9j/2wBDAAYEBQYFBAYGBQYHBwYIChAKCgkJChQODwwQFxQYGBcUFhYaHSUfGhsjHBYWICwgIyYnKSopGR8tMC0oMCUoKSj/2wBDAQcHBwoIChMKChMoGhYaKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCj/wAARCAANABQDASIAAhEBAxEB/8QAGgAAAQUBAAAAAAAAAAAAAAAAAAECAwQFBv/EACMQAAIBBAIBBQEAAAAAAAAAAAEDAgAEBRESIRMiMTJBYXH/xAAVAQEBAAAAAAAAAAAAAAAAAAAFBv/EAB4RAAICAgMBAQAAAAAAAAAAAAECAwQAMREiYXEF/9oADAMBAAIRAxEAPwDsrDGIeq0T4wHST5idda/TUN3iJQjMIUsSPx13s/yktM+5OPWiKoERhw5feqbh8tKyfNjFC4Mj0Jy1xo2v+hTSNQwGhz18+YlPRuPIxUnZ47e/cxshaSRccGq4zAGwRRV7P5A3+QLyqK/SBxB37UVL3Jked2TROU9SJ0gRX2Bn/9k=',
        },
        url: 'https://cdn.sanity.io/images/9rg5iv1m/development/bb5bb5881a12dd50347876e6a63e467bb56ba88f-1920x1222.webp',
      },
      crop: null,
      hotspot: null,
    },
    markDefs: null,
    videoUrl:
      'https://stream.mux.com/6nJQvGQWyuTlpqM7O01u4jbVgls2164Rx7MZp6QNarWo/high.mp4',
  },
  {
    _key: '12b97abf9721',
    _type: 'block',
    children: [
      {
        _key: '872a3ad088d3',
        _type: 'span',
        marks: ['textColorOne'],
        text: '#5 - Regular Video',
      },
    ],
    markDefs: [],
    style: 'headingOne',
  },
  {
    _key: '5d37e89b828d',
    _type: 'customVideoSchema',
    altDescription: 'Google Bard',
    imageFile: {
      asset: {
        metadata: {
          dimensions: {
            aspectRatio: 1.7777777777777777,
            height: 720,
            width: 1280,
          },
          lqip: 'data:image/jpeg;base64,/9j/2wBDAAYEBQYFBAYGBQYHBwYIChAKCgkJChQODwwQFxQYGBcUFhYaHSUfGhsjHBYWICwgIyYnKSopGR8tMC0oMCUoKSj/2wBDAQcHBwoIChMKChMoGhYaKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCj/wAARCAALABQDASIAAhEBAxEB/8QAGAAAAwEBAAAAAAAAAAAAAAAAAAUHAwb/xAAjEAACAQQBBAMBAAAAAAAAAAABAgQAAwUREgYTFCExQVGR/8QAFgEBAQEAAAAAAAAAAAAAAAAAAgED/8QAHBEAAQUAAwAAAAAAAAAAAAAAAQACAxEhIjFR/9oADAMBAAIRAxEAPwCLdPRjKN214neU623xxp23TLzVXwrCBVuhbiq22Uftc1PkXuaauMu0APH1v+VpgZ8qLlotyPIuI/cHsNSfbqPiUZDSc7T/AD+IxFjIG3DWQ1tVAJUEjl90VUJlw9xCAoLICdKBs0VGyChi2ki5Gl//2Q==',
        },
        url: 'https://cdn.sanity.io/images/9rg5iv1m/development/c2a2c7fd0d558f5a8a5691d1e82b25216c66163a-1280x720.jpg',
      },
      crop: null,
      hotspot: null,
    },
    markDefs: null,
    videoUrl: 'https://www.youtube.com/embed/xW9DJTvB3NI',
  },
  {
    _key: 'f302a0c77cae',
    _type: 'block',
    children: [
      {
        _key: '03771dcd113b',
        _type: 'span',
        marks: ['textColorOne'],
        text: '#6 - Code Input',
      },
    ],
    markDefs: [],
    style: 'headingOne',
  },
  {
    _key: '3424b9cc615d',
    _type: 'block',
    children: [{ _key: '97a8dea35498', _type: 'span', marks: [], text: '' }],
    markDefs: [],
    style: 'normal',
  },
  {
    _key: 'a6ef645e7c50',
    _type: 'code',
    code: "import { groq } from 'next-sanity';\nimport { client } from 'src/backend/client';\nimport { NextApiRequest, NextApiResponse } from 'next';\nimport { portableTextFragment } from 'src/backend/queries/portable-text';\n\nexport default async function fetchBlog(\n  req: NextApiRequest,\n  res: NextApiResponse\n) {\n  const query = req.query;\n  let groqQuery = '';\n\n  if ('index' in query) {\n    groqQuery = groq`\n\t\t\t*[_type == \"blog\"][${query.index}]{\n\t\t\t\t...,\n\t\t\t\tblogContent[]{\n\t\t\t\t\t${portableTextFragment}\n\t\t\t\t}\n\t\t\t}\n\t\t`;\n  } else {\n    groqQuery = groq`\n\t\t\t*[_type == \"blog\"][0]{\n\t\t\t\t...,\n\t\t\t\tblogContent[]{\n\t\t\t\t\t${portableTextFragment}\n\t\t\t\t}\n\t\t\t}\n\t\t`;\n  }\n  const blog = await client.fetch(groqQuery);\n  res.send(blog);\n}\n",
    filename: 'Sample Code Snipper',
    language: 'typescript',
    markDefs: null,
  },
  {
    _key: '12c0a768e69f',
    _type: 'block',
    children: [{ _key: 'e03153b1b9ca0', _type: 'span', marks: [], text: '' }],
    markDefs: [],
    style: 'headingThree',
  },
  {
    _key: '12c0a768e69d',
    _type: 'block',
    children: [
      {
        _key: '81b5fbabdab7',
        _type: 'span',
        marks: ['textColorOne'],
        text: 'Advantages and disadvantages of using ReactJS.',
      },
    ],
    markDefs: [],
    style: 'headingThree',
  },
  {
    _key: '77fa96d52491',
    _type: 'block',
    children: [
      {
        _key: '4757c479cd150',
        _type: 'span',
        marks: ['textColorTwo'],
        text: 'ReactJS',
      },
      {
        _key: 'aab2683d0efe',
        _type: 'span',
        marks: [],
        text: ' makes it easy to dynamically render ordered and unordered lists based on changing data. This is especially useful when the data is coming from an external source that may change frequently.',
      },
    ],
    level: 1,
    listItem: 'bullet',
    markDefs: [],
    style: 'normal',
  },
  {
    _key: '7e712654ef74',
    _type: 'block',
    children: [
      {
        _key: '34f976f1a9ee0',
        _type: 'span',
        marks: ['textColorTwo'],
        text: 'ReactJS',
      },
      {
        _key: '2069ed52d971',
        _type: 'span',
        marks: [],
        text: ' is designed to be efficient and scalable, which means that even when rendering large lists, the performance of the application will not suffer.',
      },
    ],
    level: 1,
    listItem: 'bullet',
    markDefs: [],
    style: 'normal',
  },
  {
    _key: '1eeb0be74fbc',
    _type: 'block',
    children: [
      { _key: '5e4c3a90b6fa', _type: 'span', marks: [], text: 'level-2' },
    ],
    level: 2,
    listItem: 'bullet',
    markDefs: [],
    style: 'normal',
  },
  {
    _key: '19de9eb5b966',
    _type: 'block',
    children: [
      { _key: 'f6b69f4b3250', _type: 'span', marks: [], text: 'level-2' },
    ],
    level: 2,
    listItem: 'bullet',
    markDefs: [],
    style: 'normal',
  },
  {
    _key: 'de5b81bcd215',
    _type: 'block',
    children: [
      { _key: '36e135c4acf0', _type: 'span', marks: [], text: 'level-3' },
    ],
    level: 3,
    listItem: 'bullet',
    markDefs: [],
    style: 'normal',
  },
  {
    _key: 'f4a5bad8ec3c',
    _type: 'block',
    children: [
      { _key: '0ece78050cec', _type: 'span', marks: [], text: 'level-3' },
    ],
    level: 3,
    listItem: 'bullet',
    markDefs: [],
    style: 'normal',
  },
  {
    _key: 'e60ba702e85d',
    _type: 'block',
    children: [
      { _key: 'b87bbd143753', _type: 'span', marks: [], text: 'level-4' },
    ],
    level: 5,
    listItem: 'bullet',
    markDefs: [],
    style: 'normal',
  },
  {
    _key: '23e93b20840e',
    _type: 'block',
    children: [
      { _key: 'a153036cae1c', _type: 'span', marks: [], text: 'level-4' },
    ],
    level: 5,
    listItem: 'bullet',
    markDefs: [],
    style: 'normal',
  },
  {
    _key: 'a8824b1bfa90',
    _type: 'block',
    children: [
      {
        _key: 'c580d958d5b70',
        _type: 'span',
        marks: ['textColorTwo'],
        text: 'ReactJS',
      },
      {
        _key: '40540b0da383',
        _type: 'span',
        marks: [],
        text: ' allows for easy customization and styling of ordered and unordered lists, thanks to its modular architecture and use of CSS.',
      },
    ],
    level: 1,
    listItem: 'bullet',
    markDefs: [],
    style: 'normal',
  },
  {
    _key: '8843524c1a08',
    _type: 'block',
    children: [{ _key: 'e68a7dba1498', _type: 'span', marks: [], text: '' }],
    markDefs: [],
    style: 'normal',
  },
  {
    _key: '3cf83b25b571',
    _type: 'block',
    children: [
      {
        _key: '1ff91a2605f40',
        _type: 'span',
        marks: ['textColorTwo'],
        text: 'ReactJS ',
      },
      {
        _key: '32cd4ccca7bf',
        _type: 'span',
        marks: [],
        text: 'can have a steep learning curve for beginners, especially those who are not familiar with the concepts of JavaScript, HTML, and CSS.',
      },
    ],
    level: 1,
    listItem: 'number',
    markDefs: [],
    style: 'normal',
  },
  {
    _key: 'e4f831556b83',
    _type: 'block',
    children: [
      { _key: 'c4de1bf137d8', _type: 'span', marks: [], text: 'Level-2' },
    ],
    level: 2,
    listItem: 'number',
    markDefs: [],
    style: 'normal',
  },
  {
    _key: 'f3f80f2d6506',
    _type: 'block',
    children: [
      { _key: 'b7d5b6a797ce', _type: 'span', marks: [], text: 'level-3' },
    ],
    level: 3,
    listItem: 'number',
    markDefs: [],
    style: 'normal',
  },
  {
    _key: '4e314c896197',
    _type: 'block',
    children: [
      { _key: '8c7ffc496fe6', _type: 'span', marks: [], text: 'level-3' },
    ],
    level: 3,
    listItem: 'number',
    markDefs: [],
    style: 'normal',
  },
  {
    _key: '341f49c94f1e',
    _type: 'block',
    children: [
      { _key: '8f0cc5c6fa6a', _type: 'span', marks: [], text: 'level-4' },
    ],
    level: 4,
    listItem: 'number',
    markDefs: [],
    style: 'normal',
  },
  {
    _key: '9dccbedea5ea',
    _type: 'block',
    children: [
      { _key: '2d946796d545', _type: 'span', marks: [], text: 'level-4' },
    ],
    level: 4,
    listItem: 'number',
    markDefs: [],
    style: 'normal',
  },
  {
    _key: 'd287a82aeb10',
    _type: 'block',
    children: [
      { _key: 'd53bacb442db', _type: 'span', marks: [], text: 'level-5' },
    ],
    level: 5,
    listItem: 'number',
    markDefs: [],
    style: 'normal',
  },
  {
    _key: 'c4d0cefbbd47',
    _type: 'block',
    children: [
      {
        _key: '60c5d6348f610',
        _type: 'span',
        marks: ['textColorOne'],
        text: 'ReactJS ',
      },
      {
        _key: '67c5ffb9aec5',
        _type: 'span',
        marks: [],
        text: 'can be complex to set up and requires a lot of initial configuration, which can be time-consuming.',
      },
    ],
    level: 1,
    listItem: 'number',
    markDefs: [],
    style: 'normal',
  },
  {
    _key: '17baa6b9e82c',
    _type: 'block',
    children: [
      {
        _key: '4f74190285cd0',
        _type: 'span',
        marks: ['textColorOne'],
        text: 'Because ReactJS',
      },
      {
        _key: 'a4093b18e589',
        _type: 'span',
        marks: [],
        text: ' is a JavaScript library, it may not be the best choice for developers who prefer to work with other programming languages.',
      },
    ],
    level: 1,
    listItem: 'number',
    markDefs: [],
    style: 'normal',
  },
  {
    _key: 'aa5efb05354f',
    _type: 'block',
    children: [
      { _key: '7cee13652e570', _type: 'span', marks: [], text: 'S' },
      {
        _key: '4d9a561960e7',
        _type: 'span',
        marks: ['textColorOne'],
        text: 'ince ReactJS is a client-side library, ',
      },
      {
        _key: '346b7bc9a459',
        _type: 'span',
        marks: [],
        text: 'it may not be suitable for applications that require server-side rendering, as this can be challenging to implement.',
      },
    ],
    level: 1,
    listItem: 'number',
    markDefs: [],
    style: 'normal',
  },
];

export const customPortableTextProps: ICustomPortableText = {
  content: richTextContent,
  isDefaultBg: true,
};
