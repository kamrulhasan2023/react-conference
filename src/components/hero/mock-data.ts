import { IRCHero } from './interface';
export const heroProps: IRCHero = {
    title: "React Conference",
    description: "Lorem uis diam turpis quam id fermentum.In quis diam turpis quam id fermentum..id fermentum.In quis diam turpi quam id fermentum.",
    btnText: "Buy Tickets",
};