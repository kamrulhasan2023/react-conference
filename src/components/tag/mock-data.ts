import { ITag } from './interface';

export const tagProps: ITag = {
  label: 'Nextjs',
  color: 'bg-danger',
};
