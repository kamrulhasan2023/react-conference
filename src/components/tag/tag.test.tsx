import { render } from '@testing-library/react';
import Tag from '.';
import { tagProps } from './mock-data';

it('renders tag correctly', () => {
  const { getByRole } = render(<Tag {...tagProps} />);
  const element = getByRole('presentation');
  expect(element).toBeInTheDocument();
  expect(element).toMatchSnapshot();
});
