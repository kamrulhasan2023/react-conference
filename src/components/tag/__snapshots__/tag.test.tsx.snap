// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders tag correctly 1`] = `
<span
  class="inline-flex items-center justify-center px-3 py-1 text-sm font-semibold text-black rounded-full bg-danger"
  role="presentation"
>
  Nextjs
</span>
`;
